scripts/nightly_bench.py

```
@@ -0,0 +1,535 @@
+        return 1, 5
+    return 1, 3
+
+
+def calibrate_timeout() -> None:
+    """Decide whether wrapping runs in `timeout` is cheap enough to use.
+
+    GNU coreutils `timeout` adds ~1ms; some reimplementations add ~100ms, which
+    would dominate fast benchmarks. We only use the wrapper when its overhead is
+    negligible; either way the per-run cap is also enforced by probe() and by a
+    subprocess-level backstop in hyperfine()."""
+    global TIMEOUT_PREFIX
```

Ruff (0.15.18)

[warning] Using the global statement to update `TIMEOUT_PREFIX` is discouraged

(PLW0603)